libs/net/dal/Migrations/1.0.156/Up/PostUp/00-ChartTemplate-CoverageCount.sql

```diff
@@ -0,0 +1,93 @@
+    Dictionary<string, List<int>> mentionsByMediaType = new Dictionary<string, List<int>>();
+    allMediaTypes.ForEach(mt => mentionsByMediaType.Add(mt, new List<int>()));
+
+    // Get all groups in data
+    var groups = Content
+      .GroupBy(
+        c => SectionSettings.GroupBy switch
+        {
+          "contentType" => c.ContentType.ToString(),
+          "byline" => c.Byline,
+          "series" => c.Series?.Name,
+          "source" => c.OtherSource,
+          "dayMonthYear" => $"{c.PublishedOn:dd-MM-yyyy}",
+          "monthYear" => $"{c.PublishedOn:MM-yyyy}",
+          "year" => $"{c.PublishedOn:yyyy}",
+          "sentiment" => $"{c.TonePools.FirstOrDefault()?.Value ?? 0}",
+          "sentimentSimple" => (c.TonePools.FirstOrDefault()?.Value ?? 0) < 0 ? "Negative" : (c.TonePools.FirstOrDefault()?.Value ?? 0) > 0 ? "Positive": "Neutral",
+          "reportSection" => c.SectionName,
+          _ => c.OtherSource,
+        },
+        c => c,
+        (k, c) => {
+          Dictionary<string, int> groupMentionsByMediaType = c.GroupBy(g => g.MediaType?.Name ?? UnknownMediaTypeLabel)
+            .ToDictionary( gdc => gdc.Key, gdc => gdc.Count());
+        
+          // iterate over each Media Type that was found in the collective Content
+          foreach(string mediaType in allMediaTypes ) {
+            // if this grouping contains the current Media Type, add it to the master list
+            if (groupMentionsByMediaType.ContainsKey(mediaType)) {
+              mentionsByMediaType[mediaType].Add(groupMentionsByMediaType[mediaType]);
+            } else {
+              // no mention for this Media Type in this grouping, pad the array with a 0
+              mentionsByMediaType[mediaType].Add(0);
+            }
+          }
+
+        return new {
+          Label = SectionSettings.GroupBy != "reportSection" ? k : Sections.ContainsKey(k) ? Sections[k].Settings.Label : "",
+          MentionsByMediaType = groupMentionsByMediaType
+        };
+      })
+      .OrderBy(g=> g.Label)
+      .ToArray();
+      
+      // build up the objects to serialize for the chart data
+     var labels = groups.Select(value => value.Label);
+     var datasets = mentionsByMediaType.Select(m => 
+       new {
+         label = m.Key
+         ,data = m.Value
+         //,backgroundColor = TODO: map Media Type to fixed color
+       }
+     );
+}
+@* now render the JSON for the chart data*@
+{
+  "labels": @JsonSerializer.Serialize(labels) ,
+  "datasets": @JsonSerializer.Serialize(datasets) 
+   @* ,"rawContent": @.JsonSerializer.Serialize(Content) *@
+   @* ,"rawGroups": @JsonSerializer.Serialize(groups) *@
+   @* ,"rawSections": @JsonSerializer.Serialize(Sections)  *@
+   @* ,"rawMentionsByMediaType": @JsonSerializer.Serialize(mentionsByMediaType) *@
+   @* ,"rawDatasets": @JsonSerializer.Serialize(datasets) *@
+}' -- template
```

Comment:
deliberately left this here as a guide for some poor soul trying to build a new chart template